src/backend/distributed/deparser/ruleutils_13.c

```diff
@@ -4881,6 +4881,12 @@ get_rule_expr(Node *node, deparse_context *context,
 	CHECK_FOR_INTERRUPTS();
 	check_stack_depth();
 
+	if (nodeTag(node) == T_RelabelType &&
+		((RelabelType *)node)->resultcollid != InvalidOid)
```

Comment:
why not do this in the switch statement below?

Reply:
It didn't work. I guess the reason is: Switch statement checks the initial node only; however, the node we are looking for might be somewhere we can reach by recursive search. @halilozanakgul what do you think?

Reply:
I don't follow.

How can
```c
if (nodeTag(node) == T_RelabelType && ((RelabelType *) node)->resultcollid != InvalidOid)
{
    node = RelabelTypeMutator(node);
}

switch (nodeTag(node))
{
    case T_RelabelType:
    {
        ...
    }
}
```

be different from 
```c

switch (nodeTag(node))
{
    case T_RelabelType:
    {
        RelabelType *relabel = (RelabelType *) node;
        Node	   *arg = (Node *) relabel->arg;

        if (resultcollid != InvalidOid)
        {
            node = RelabelTypeMutator(node);
        }
        ...
    }
}

``` 
?

Reply:
ah, you're right. I've misunderstood

Reply:
Ah, because in the first case we go into the CollateExpr switch I guess.

Reply:
what if we do:
```c
case T_RelabelType:
{
    /*
     * The planner rewrites CollateExpr to RelabelType, hence we print
     * RelabelType with a collation ID as a COLLATE expression.
     */
    if (relabel->resultcollid != InvalidOid)
    {
        if (!PRETTY_PAREN(context))
            appendStringInfoChar(buf, '(');
        get_rule_expr_paren(arg, context, showimplicit, node);
        appendStringInfo(buf, " COLLATE %s",
                         generate_collation_name(collate->collOid));
        if (!PRETTY_PAREN(context))
            appendStringInfoChar(buf, ')');
    }
}
```